addons/web/controllers/main.py

```diff
@@ -818,10 +818,19 @@ def _write_group_header(self, row, column, label, group, group_depth=0):
         for field in self.fields[1:]: # No aggregates allowed in the first column because of the group title
             column += 1
             aggregated_value = aggregates.get(field['name'])
-            # Non-stored float fields may not be displayed properly because of float representation
-            # => we force 2 digits
-            if not field.get('store') and isinstance(aggregated_value, float):
-                aggregated_value = float_repr(aggregated_value, 2)
+            # Float fields may not be displayed properly because of float
+            # representation issue with non stored fields or with values
+            # that, even stored, cannot be rounded properly and it is not
+            # acceptable to display useless digits (i.e. monetary)
+            #
+            # non stored field ->  we force 2 digits
+            # stored monetary -> we force max digits of installed currencies
+            if isinstance(aggregated_value, float):
+                if field.get('type') == 'monetary' and hasattr(group._model, 'currency_id'):
```

Comment:
```suggestion
                if field.get('type') == 'monetary':
```

A monetary field is always linked to a currency, so I don't think this is necessary.